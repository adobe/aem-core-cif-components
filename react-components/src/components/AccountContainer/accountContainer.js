/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { useTranslation, Trans } from 'react-i18next';

import { useUserContext } from '../../context/UserContext';
import AccountTrigger from './accountTrigger';
import AccountDropdown from './accountDropdown';

const AccountContainer = () => {
    const [{ currentUser, isSignedIn }] = useUserContext();
    const [t] = useTranslation('account');

    const label = isSignedIn ? (
        <Trans t={t} i18nKey="account:account-icon-text-greeting">
            Hi, {{ name: currentUser.firstname }}
        </Trans>
    ) : (
        t('account:account-icon-text-sign-in', 'Sign In')
    );

    return (
        <>
            <AccountTrigger label={label} />
            <AccountDropdown />
        </>
    );
};

export default AccountContainer;
