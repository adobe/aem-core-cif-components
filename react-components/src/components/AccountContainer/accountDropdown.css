.root {
    background-color: rgb(var(--venia-grey));
    display: grid;
    gap: 1rem;
    min-width: 22rem;
    left: 50%;
    opacity: 0;
    position: absolute;
    top: 3.3rem;
    transform: translate3d(-50%, 0, 0);
    transition-duration: 192ms;
    transition-property: opacity, transform, visibility;
    transition-timing-function: cubic-bezier(0.4, 0, 1, 1);
    visibility: hidden;
    z-index: 4;
}

.root_open {
    composes: root;
    opacity: 1;
    transition-duration: 224ms;
    transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
    visibility: visible;
}

.mask_active {
    composes: root_active from '../Mask/mask.css';
    opacity: 0;
}
