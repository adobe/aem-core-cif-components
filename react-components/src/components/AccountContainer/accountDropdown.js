/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';

import { useConfigContext } from '../../context/ConfigContext';
import { useUserContext } from '../../context/UserContext';
import SignIn from '../SignIn/signIn';
import MyAccount from '../MyAccount/myAccount';
import ChangePassword from '../ChangePassword';
import ForgotPassword from '../ForgotPassword/forgotPassword';
import CreateAccount, { CreateAccountSuccess } from '../CreateAccount';

import classes from './accountDropdown.css';
import Mask from '../Mask';

const AccountDropdown = () => {
    const [
        { isAccountDropdownOpen, isSignedIn, accountDropdownView },
        {
            showSignIn,
            showMyAccount,
            showForgotPassword,
            showCreateAccount,
            showAccountCreated,
            showChangePassword,
            toggleAccountDropdown
        }
    ] = useUserContext();
    const { pagePaths } = useConfigContext();
    const rootClass = isAccountDropdownOpen ? classes.root_open : classes.root;

    let view;
    if (!isSignedIn) {
        switch (accountDropdownView) {
            case 'FORGOT_PASSWORD':
                view = <ForgotPassword onClose={showSignIn} onCancel={showSignIn} />;
                break;
            case 'CREATE_ACCOUNT':
                view = (
                    <CreateAccount
                        showMyAccount={showMyAccount}
                        showAccountCreated={showAccountCreated}
                        handleCancel={showSignIn}
                    />
                );
                break;
            case 'ACCOUNT_CREATED':
                view = <CreateAccountSuccess showSignIn={showSignIn} />;
                break;
            case 'SIGN_IN':
            default:
                view = (
                    <SignIn
                        showMyAccount={showMyAccount}
                        showForgotPassword={showForgotPassword}
                        showCreateAccount={showCreateAccount}
                    />
                );
        }
    } else {
        switch (accountDropdownView) {
            case 'CHANGE_PASSWORD':
                view = <ChangePassword showMyAccount={showMyAccount} handleCancel={showMyAccount} />;
                break;
            case 'MY_ACCOUNT':
            default:
                view = (
                    <MyAccount
                        showChangePassword={showChangePassword}
                        showAccountInformation={() => {
                            window.location.href = pagePaths.accountDetails;
                        }}
                    />
                );
        }
    }

    return (
        <>
            <Mask
                isOpen={isAccountDropdownOpen}
                onClickHandler={() => toggleAccountDropdown(!isAccountDropdownOpen)}
                customClasses={{ root_active: classes.mask_active }}
            />

            <div className={rootClass} aria-label="account dropdown">
                {view}
            </div>
        </>
    );
};

export default AccountDropdown;
