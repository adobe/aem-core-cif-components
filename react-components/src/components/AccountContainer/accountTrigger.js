/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import PropTypes from 'prop-types';
import { User as UserIcon } from 'react-feather';

import { useUserContext } from '../../context/UserContext';
import Icon from '../Icon';

import classes from './accountTrigger.css';

const AccountTrigger = props => {
    const [{ isAccountDropdownOpen }, { toggleAccountDropdown }] = useUserContext();
    const { label } = props;

    const iconColor = 'rgb(var(--venia-text))';
    const svgAttributes = {
        stroke: iconColor
    };

    return (
        <button
            className={classes.root}
            aria-label="Toggle account dropdown"
            onClick={() => toggleAccountDropdown(!isAccountDropdownOpen)}>
            <Icon src={UserIcon} attrs={svgAttributes} />
            {label && <span className={classes.label}>{label}</span>}
        </button>
    );
};

AccountTrigger.propTypes = {
    children: PropTypes.node,
    classes: PropTypes.shape({
        root: PropTypes.string
    }),
    label: PropTypes.oneOfType([PropTypes.object, PropTypes.string])
};

export default AccountTrigger;
