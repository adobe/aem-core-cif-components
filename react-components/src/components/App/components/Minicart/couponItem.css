.root {
    display: grid;
    grid-gap: 0 1rem;
    grid-template-columns: 1fr min-content;
    margin: 0 1.5rem;
}
.couponName {
    line-height: 1.5rem;
}
