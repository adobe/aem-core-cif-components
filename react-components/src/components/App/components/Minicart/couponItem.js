/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { useCartState } from './cartContext';
import Kebab from './kebab';
import Section from './section';

import classes from './couponItem.css';

const CouponItem = () => {
    const [{ cart, removeCoupon }] = useCartState();

    const appliedCoupon = cart.applied_coupon ? cart.applied_coupon.code : null;

    return (
        <div className={classes.root}>
            <div className={classes.couponName}>
                Coupon <strong>{appliedCoupon}</strong> applied.
            </div>
            <Kebab>
                <Section text="Remove coupon" onClick={removeCoupon} icon="Trash" />
            </Kebab>
        </div>
    );
};

export default CouponItem;
