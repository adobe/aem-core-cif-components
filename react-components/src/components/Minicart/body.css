.root {
    display: grid;
    grid-template-rows: min-content min-content 1fr;
    overflow: auto;
}
