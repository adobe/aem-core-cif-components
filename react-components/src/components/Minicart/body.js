/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { useTranslation } from 'react-i18next';

import LoadingIndicator from '../LoadingIndicator';

import EmptyMinicartBody from './emptyMinicartBody';
import classes from './body.css';
import ProductList from './productList';
import CartOptions from './cartOptions';
import Error from './error';
import CouponForm from './couponForm';
import CouponItem from './couponItem';

import { useCartState } from './cartContext';
import DiscountList from './discountList';

const CartLoadingIndicator = () => {
    const [t] = useTranslation('cart');

    return <LoadingIndicator>{t('cart:fetching-data', 'Fetching cart data...')}</LoadingIndicator>;
};

const Body = () => {
    const [{ isEditing, cart, isLoading, errorMessage }] = useCartState();
    const isEmpty = cart && Object.entries(cart).length > 0 ? cart.items.length === 0 : true;

    if (isLoading) {
        return <CartLoadingIndicator />;
    }

    if (errorMessage) {
        return <Error />;
    }

    if (isEmpty) {
        return <EmptyMinicartBody />;
    }

    if (isEditing) {
        return <CartOptions />;
    }

    const cartItems = cart.items;
    const couponFragment = !cart.applied_coupon ? <CouponForm /> : <CouponItem />;
    const discountFragment =
        cart.prices.discounts && cart.prices.discounts.length > 0 ? (
            <DiscountList discounts={cart.prices.discounts} />
        ) : (
            <div />
        );

    return (
        <div className={classes.root}>
            <ProductList cartItems={cartItems} />
            {discountFragment}
            {couponFragment}
        </div>
    );
};

export default Body;
