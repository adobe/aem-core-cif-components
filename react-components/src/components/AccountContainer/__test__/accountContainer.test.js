/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { render } from 'test-utils';
import { CartProvider } from '../../Minicart';
import AccountContainer from '../accountContainer';

describe('<AccountContainer>', () => {
    it('renders the component', () => {
        const config = {
            graphqlEndpoint: 'endpoint',
            storeView: 'default',
            pagePaths: {
                accountDetails: '/accountDetails'
            }
        };

        const { asFragment } = render(
            <CartProvider initialState={{ cartId: null }} reducerFactory={() => state => state}>
                <AccountContainer />
            </CartProvider>,
            { config: config }
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
