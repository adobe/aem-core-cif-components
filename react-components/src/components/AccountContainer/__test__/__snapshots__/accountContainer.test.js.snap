// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountContainer> renders the component 1`] = `
<DocumentFragment>
  <button
    aria-label="Toggle account dropdown"
    class="root"
  >
    <span
      class="root"
    >
      <svg
        fill="none"
        height="24"
        stroke="rgb(var(--venia-text))"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
        />
        <circle
          cx="12"
          cy="7"
          r="4"
        />
      </svg>
    </span>
    <span
      class="label"
    >
      Sign In
    </span>
  </button>
  <button
    data-role="mask"
  />
  <div
    aria-label="account dropdown"
    class="root"
  >
    <div
      class="root"
    >
      <form
        class="form"
      >
        <div
          class="formTitle"
        >
          Sign-in to Your Account
        </div>
        <div
          class="root"
        >
          <span
            class="label"
          >
            <span
              class="requiredSymbol"
            />
            Email address
          </span>
          <span
            class="root"
          >
            <span
              class="input"
            >
              <input
                aria-label="email"
                autocomplete="email"
                class="input"
                name="email"
                value=""
              />
            </span>
            <span
              class="before"
            />
            <span
              class="after"
            />
          </span>
          <p
            class="root"
          />
        </div>
        <div
          class="root"
        >
          <span
            class="label"
          >
            <span
              class="requiredSymbol"
            />
            Password
          </span>
          <span
            class="root"
          >
            <span
              class="input"
            >
              <input
                aria-label="password"
                autocomplete="current-password"
                class="input"
                name="password"
                type="password"
                value=""
              />
            </span>
            <span
              class="before"
            />
            <span
              class="after"
            />
          </span>
          <p
            class="root"
          />
        </div>
        <div
          class="forgotPasswordButton"
        >
          <button
            class="root_lowPriority"
            type="button"
          >
            <span
              class="content"
            >
              Forgot Password?
            </span>
          </button>
        </div>
        <div
          class="signInError"
        />
        <div
          class="signInButton"
        >
          <button
            aria-label="submit"
            class="root_highPriority"
            type="submit"
          >
            <span
              class="content"
            >
              Sign In
            </span>
          </button>
        </div>
        <div
          class="createAccountButton"
        >
          <button
            class="root_normalPriority"
            type="button"
          >
            <span
              class="content"
            >
              Create an Account
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
